source/common/http/http2/codec_impl.cc

```diff
@@ -608,8 +617,12 @@ void ConnectionImpl::ServerStreamImpl::decodeTrailers() {
   // Consume any buffered trailers.
   stream_manager_.trailers_buffered_ = false;
 
-  request_decoder_->decodeTrailers(
-      std::move(absl::get<RequestTrailerMapPtr>(headers_or_trailers_)));
+  RequestDecoder* request_decoder = request_decoder_handle_->get().ptr();
```

Comment:
Can we just call decoder() method here?  like RequestDecoder* request_decoder = decoder();

Reply:
Good catch, done!

Reply:
Actually the reason I did that is decoder() returns a StreamDecoder* which doesn't have methods to decode headers or trailers, hence why I directly get the RequestDecoder*.